core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -640,13 +642,15 @@ public void operationComplete(ChannelFuture f) throws Exception {
                     primaryActivePort = actualPort;
                 }
 
-                if (localAddress.getAddress().isAnyLocalAddress() ||
-                    localAddress.getAddress().isLoopbackAddress()) {
-                    logger.info("Serving {} at {} - {}://127.0.0.1:{}/",
-                                port.protocol().name(), localAddress,
-                                port.protocol().uriText(), localAddress.getPort());
-                } else {
-                    logger.info("Serving {} at {}", port.protocol().name(), localAddress);
+                if (logger.isInfoEnabled()) {
+                    if (localAddress.getAddress().isAnyLocalAddress() ||
+                        localAddress.getAddress().isLoopbackAddress()) {
+                        port.protocols().forEach(p -> logger.info(
+                                "Serving {} at {} - {}://127.0.0.1:{}/",
+                                p.name(), localAddress, p.uriText(), localAddress.getPort()));
+                    } else {
+                        logger.info("Serving {} at {}", Joiner.on('+').join(port.protocols()), localAddress);
```

Comment:
`String.join`

Reply:
`SessionProtocol` is not a `CharSequence`.